spring-boot-project/spring-boot/src/main/java/org/springframework/boot/jackson/JsonObjectDeserializer.java

```diff
@@ -117,18 +118,4 @@ protected final <D> D nullSafeValue(JsonNode jsonNode, Class<D> type) {
 		}
 		throw new IllegalArgumentException("Unsupported value type " + type.getName());
 	}
-
-	/**
-	 * Helper method to return a {@link JsonNode} from the tree.
-	 * @param tree the source tree
-	 * @param fieldName the field name to extract
-	 * @return the {@link JsonNode}
-	 */
-	protected final JsonNode getRequiredNode(JsonNode tree, String fieldName) {
-		Assert.notNull(tree, "Tree must not be null");
-		JsonNode node = tree.get(fieldName);
-		Assert.state(node != null && !(node instanceof NullNode), () -> "Missing JSON field '" + fieldName + "'");
-		return node;
-	}
-
```

Comment:
We don't order methods by scope (which I assume is why you moved this), instead we try to [arrange them in the order that you would read them when reviewing the code](https://github.com/spring-io/spring-javaformat?tab=readme-ov-file#read-down-methods-fields-and-parameters). Please move it back to the bottom of the class.